explanation/journey-of-a-change-to-production.rst

```diff
@@ -0,0 +1,317 @@
+=========================================================================
+The journey of a Launchpad change from a developer's branch to production
+=========================================================================
+
+.. warning::
+
+   This document contains many links that are Canonical-internal and only
+   accessible to those who work for Canonical.
+
+##########
+Characters
+##########
+
+* The Launchpad developer.
+* Launchpad's branches.
+* Launchpad itself.
+* Jenkins jobs.
+* Buildbot.
+* Deployable.
+* Launchpad ``qastaging`` and ``staging`` environments.
+* Scripts from the ``lp:launchpad-bastion-scripts`` repository.
+* Juju and mojo.
+* Fragile Launchpad jobs.
+* Launchpad production environment.
+
+###########
+The journey
+###########
+
+What VCS branches does Launchpad use?
+-------------------------------------
+
+The `main Launchpad repository <https://git.launchpad.net/launchpad>`_ has 4
+main (or "trunk") branches. These are ``master``, ``stable``, ``db-devel``,
+and ``db-stable``. For more details see `branches <../branches>`_.
+
+How to contribute a change to Launchpad?
+----------------------------------------
+
+If the change includes code and database schema changes, both have to be done
+separately and target different branches. The database patch needs to be based
+on and target the ``db-devel`` branch, whereas the code changes need to be based
+on and target the ``master`` branch. This is because database changes can be
+destabilising to other work, so we isolate them into a separate branch. Since
+Launchpad has a large database and codebase with strict uptime requirements,
+we want to deploy changes to only one of these at a time and this separation
+helps that as well.
+
+How to test the code and database changes together?
+---------------------------------------------------
+
+It is important to get the database patches submitted and merged to the
+``db-devel`` branch as early as possible. This is because those changes get
+merged to the ``master`` branch only after they are deployed to production
+(the details will be explained in a later question) and it is a pre-requisite
+to get the code changes merged.
+
+While the database patches make their way to production and to the ``master``
+branch, it is okay to copy the database patch from the ``db-devel``
+branch manually and apply it locally in the development environment to
+implement and test the code changes.
+
+How to deploy the database changes to the production environment?
+-----------------------------------------------------------------
+
+The database patches should be submitted to the ``db-devel`` branch and once
+they are approved, they get merged by a landing (aka "merging") bot which
+performs the merge. The trunk branches of Launchpad have ACLs to only allow
```

Comment:
Perhaps:

> The main Launchpad repository have ACLs so that only the merge bot is allowed to push changes to trunk branches.

The current sentence sounds ambiguous, it can be interpreted as 'the buildbot only has permissions to merge to trunk branches (as in no other branches)'.